### ISSUE TITLE
Actions/filters, register, login, logout - the whole shabang

### ISSUE THREAD

**Issue by multiplehats** [thatplugincompany/login-designer#16]
Hey man!

So, I've been using the [Theme My Login](https://wordpress.org/plugins/theme-my-login/) plugin. Which is just horribly maintained and my SCSS stylesheet and functions are piling up quickly just from this plugin. 

But the reason I'm using it is because I have several community sites that need a clean login and register on the frontend. Whether that's through a shortcode or copying over a template to the child theme.  

### Example of one of my sites
![Image of register](https://storage.jumpshare.com/preview/g_KJZVaBS3arppPmkjd4zxpd6TCQNepW9iAol0vs4Xqbon8oo0dMipsNf0CvONeI9GTLWXZWGT33s-0R1nqrFN0Iq-_ZMIwlJNqsu6s4bO0F1kR3dMUjedqC16uBUu85)

### Actions / filters
Also the ability to use a set of actions / filters would help. An example use case that I currently struggle with is: User adds product to _Checkout_ -> User needs to _register_ to use the Woo _Checkout_ so is being redirect to the  _Register Page_ -> The _Register Page_ recognises the referrer after registration and sends user back to the  _Checkout_ to complete the purchase.

### TL;DR
I got the above working, but it's all just really hacky and I don't like that in production.  

I guess my question is: are there any plans to:
- Enable a front-end login / register through a shortcode or template override.
- Have control over this with actions / filters (an example use case above)
- Ability to easily override the CSS. A lot of plugins hardcode styles and I constantly have to use `!important` (you don't strike me as this guy though)

I have probably more ideas. But it's crazy how something "simple" like this is hard to find and that's compatible with something as widely used as WooCommerce. I guess the other route would be to customize Woo's login and use that for front-end login — which probably is another deep rabbit hole.

Anyhoo, let me know what you think 👋

**Comment by richtabor** [comment]
Hey @chrisschwartze, Good to hear from you!

At this moment, I don't know if I really want to go down the shortcode route for front-end registration/login pages. I personally prefer to lean on the WordPress core functionality, and essentially "add" to the page, instead of trying to replicate it. That's just a lot of maintenance and upkeep — not to mention a lot of responsibility. Also, if I can keep using the core login functionality, any filters/actions applied there — should theoretically work within my plugin. 

I do have plans to release login/logout redirection settings (available within the Settings panel of the Login Designer Customizer instance), and possibly even login URL renaming (wp-login.php). 

For styling, I'm trying to build Login Designer so that all styling is enabled through the Customizer settings. I try my best not to be so specific in styling that folks have to jump through hoops to override something — but 9.5 times out of 10, it's already a built-in style option. 👍
